src/main/java/net/modfest/scatteredshards/api/impl/ShardLibraryPersistentState.java

```diff
@@ -0,0 +1,113 @@
+package net.modfest.scatteredshards.api.impl;
+
+import net.minecraft.nbt.NbtCompound;
+import net.minecraft.nbt.NbtList;
+import net.minecraft.nbt.NbtString;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.util.Identifier;
+import net.minecraft.world.PersistentState;
+import net.modfest.scatteredshards.ScatteredShards;
+import net.modfest.scatteredshards.api.ScatteredShardsAPI;
+import net.modfest.scatteredshards.api.ShardLibrary;
+import net.modfest.scatteredshards.api.shard.Shard;
+import net.modfest.scatteredshards.api.shard.ShardType;
+
+public class ShardLibraryPersistentState extends PersistentState {
+	public static PersistentState.Type<ShardLibraryPersistentState> TYPE = new PersistentState.Type<>(
+			ShardLibraryPersistentState::new,
+			ShardLibraryPersistentState::createFromNbt,
+			null
+			);
+	
+	public static final String SHARD_TYPES_KEY = "ShardTypes";
+	public static final String SHARDS_KEY = "Shards";
+	public static final String SHARD_SETS_KEY = "ShardSets";
+	
+	public static ShardLibraryPersistentState get(MinecraftServer server) {
+		ShardLibraryPersistentState result = server.getOverworld().getPersistentStateManager().getOrCreate(TYPE, ScatteredShards.ID);
```

Comment:
maybe use as `scattered_shards_library` like with collection instead of just `scattered_shards`